### DISCUSSION TITLE
请问该如何正确加载用户通过input[type=file]选择的模型文件？

### DISCUSSION THREAD

**Discussion by Jarod-Xie** [galacean/discussion#31] [Engine] [answered]
# 我的思路
获取到File对象后，通过
```
    var reader = new FileReader();
    reader.readAsText(file);
    reader.onload = function () {
      //当读取完成之后会回调这个函数，然后此时文件的内容存储到了result中。

      engine.resourceManager.load<GLTFResource>(this.result);
    };
```
但是会解析错误
![image](https://user-images.githubusercontent.com/27668637/162566840-709df9a9-f1fa-4110-a87a-e79b607c3ba5.png)
请问该如何正确加载呢？

**Comment by gz65555** [comment]
@zhuxudong 后面把 glTFParser export 出去才可以吧

**Comment by zhuxudong** [comment]
内置的 parser 也不支持解析 file，可以先把file 转成blob url 解决

**Comment by gz65555** [comment]
file 转成 json 更直接一点吧

**Comment by Jarod-Xie** [comment, marked as answer]
> 内置的 parser 也不支持解析 file，可以先把file 转成blob url 解决

好的，学习官方的代码解决了。https://github.com/oasis-engine/oasis-engine.github.io/blob/main/src/pages/gltf-viewer.tsx

**Comment by eyworldwide** [comment]
@zhuxudong 这个功能做一些本地工具挺常用，加个 playground 吧
